wcfsetup/install/files/lib/data/article/category/ArticleCategory.class.php

```diff
@@ -195,6 +196,27 @@ public function getLabelGroups(string $permission = 'canSetLabel'): array
         return [];
     }
 
+    /**
+     * Returns the group of all label pickers that have been assigned to one or
+     * more accessible categories.
+     *
+     * @since 6.1
+     */
+    public static function getLabelPickerGroup(): LabelPickerGroup
+    {
+        $labelGroupsToCategories = ArticleCategoryLabelCacheBuilder::getInstance()->getData();
+        $accessibleCategoryIDs = self::getAccessibleCategoryIDs();
+
+        $groupIDs = [];
+        foreach ($labelGroupsToCategories as $categoryID => $__groupIDs) {
+            if (\in_array($categoryID, $accessibleCategoryIDs)) {
+                $groupIDs = \array_merge($groupIDs, $__groupIDs);
+            }
+        }
+
+        return LabelPickerGroup::fromGroupIDs(\array_unique($groupIDs), false);
```

Comment:
Is it ever useful to handle duplicate `groupIDs`? If not, I'd push the `array_unique` call down into `fromGroupIDs()` to avoid repeating (and forgetting it) everywhere.